tests/queries/0_stateless/02374_analyzer_array_join.reference

```diff
@@ -82,12 +82,12 @@ SELECT id, value, value_array, value_array_element FROM test_table ARRAY JOIN va
 SELECT '--';
 --
 SELECT id, value, value_array AS value_array_array_alias FROM test_table ARRAY JOIN value_array_array_alias;
-0	Value	[1,2,3]
-0	Value	[1,2,3]
-0	Value	[1,2,3]
-0	Value	[4,5,6]
-0	Value	[4,5,6]
-0	Value	[4,5,6]
+0	Value	1
```

Comment:
Ok. This is incompatible.
But the query is incorrect if we follow the logic of JOIN.
And current behavior is probably more logical.

Anyway, if you know how to add a compatibility flag in a simple way - let's add

Reply:
Yes, this is a little bit incompatible change, but I have absolutely no idea how to support the previous behavior. :(
Do you think we should mark the PR as backward-incompatible? I'd like to try to backport these changes.